### PR TITLE
fix: Make Rust work better in Bazel

### PR DESCRIPTION
**Pull request by martijneken**
* switch from cargo-raze to crate_universe
* target WASI platform
* optimize for execution speed
* add logging to stdout for unit tests
* [cleanup] set edition/crate-type to match rust_rules 

- [x] Tests pass
- [x] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: test/framework.cc [hunk 1]
```diff
@@ -33,6 +33,11 @@ uint64_t TestContext::getMonotonicTimeNanoseconds() {
   unimplemented();
   return 0;
 }
+proxy_wasm::WasmResult TestContext::log(uint32_t log_level,
+                                        std::string_view message) {
+  std::cout << "LOG from testcontext: " << message << std::endl;
```
Comment (leonm1) [inline test/framework.cc:38]: Did you intend to leave this in here? Fine if so, just wanted to call attention in case it was not

**Review by leonm1 — APPROVED** [review]
Much cleaner than having a whole tree of cargo-raze generated files!

File: test/framework.cc (hunk 1 above)
Comment (martijneken) [reply test/framework.cc:38]: Yep thanks, not really Rust specific but I found myself stashing and reinstating this over and over while working on plugins.

**Comment by mpwarres** [comment]
(Sorry for late response) +1, looks like a great simplification!

**Pull request merged**